### PR TITLE
Big Documentation Clean Up

### PR DESCRIPTION
**Pull request by sdolemelipone**
This is a clean up and expansion of the documentation. It's the first time I've done any restructuring of docs so please let me know if you see any improvements needed. Maybe the formset views section is so large that they should be divided up into more manageable pages.

I didn't modify the ListView documentation much as I'm not sure what their future is.

### CODE REVIEW

**Comment by sdolemelipone** [comment]
One of the deliberate changes I made was to reduce the prominence of GenericInline formsets in documentation as this sometimes seemed to confuse new people who think "yeah, I'd like a generic view".

**Comment by sdolemelipone** [comment]
I'd also add that I think all issues flagged as `documentation` can be closed after this with the exception of #52, which still needs completing.

**Comment by jonashaag** [comment]
I'll need a few days to review this. Does RTD support PRs? Otherwise I'll have to build it locally.

**Comment by sdolemelipone** [comment]
It only supports branches/tags on the main repo as far as I can see. The other option would be to create a new branch, switch the target of my pull request to that branch and accept the PR there, then review on RTD.

That might be a good idea just because I haven't seen how it builds on RTD, only locally.

**Comment by jonashaag** [comment]
I wanted to give you commit permission to the branch but I don't have admin permissions, @AndrewIngram can you give me admin?

**Comment by jonashaag** [comment]
I pushed it as `pr-178` branch

**Comment by AndrewIngram** [comment]
I don't think I can control collaborator permissions without moving the repo over to an organisation, but i've sent an invite to @sdolemelipone

File: README.rst [hunk 1]
```diff
@@ -1,110 +1,53 @@
-|travis| |codecov|
+|travis| |codecov| |docs-status|
 
-django-extra-views - The missing class-based generic views for Django
+Django Extra Views - The missing class-based generic views for Django
 ========================================================================
 
-Django's class-based generic views are great, they let you accomplish a large number of web application design patterns in relatively few lines of code.  They do have their limits though, and that's what this library of views aims to overcome.
+Django-extra-views is a Django app which introduces additional class-based views
```
Comment (auvipy) [inline README.rst]: app or package/extension?

File: extra_views_tests/models.py [hunk 2]
```diff
@@ -39,6 +40,15 @@ def __unicode__(self):
         return '%s (%s)' % (self.name, self.sku)
 
 
+class Contact(models.Model):
+    name = models.CharField(max_length=255)
+    email = models.CharField(max_length=255)
+    order = models.ForeignKey(Order, related_name='contacts', on_delete=models.CASCADE)
+
+    def __unicode__(self):
```
Comment (auvipy) [inline extra_views_tests/models.py]: __str__(self): should be better as python 3 is the currently supported version of django.

File: README.rst (hunk 1 above)
Comment (sdolemelipone) [reply README.rst]: Agreed!
```suggestion
Django-extra-views is a Django package which introduces additional class-based views
```

**Comment by sdolemelipone** [comment]
@auvipy thanks for the review!

**Comment by sdolemelipone** [comment]
@jonashaag I'm just waiting for the new `pr-178` branch to appear on read the docs then we can take a look at this there as well...I imagine there is a delay in the site picking up a new active branch.

**Comment by sdolemelipone** [comment]
RTD version available here: https://django-extra-views.readthedocs.io/en/pr-178/

**Comment by jonashaag** [comment]
Great stuff!! Feel free to merge anytime.

Maybe we could add a very short list of features in the intro:

- `FormSetView`, `ModelFormSetView`: Add a single formset to a page.
- `CreateWithInlinesView`, `UpdateViewInlinesView`: Create and edit models with inlines.
- `SearchableListView`: Add `?q=...` filtering to a `ListView`.
- etc.

**Comment by sdolemelipone** [comment]
Thanks, I'll add that in and merge.
After that may I go through and close the documentation issues which this addresses?

**Comment by jonashaag** [comment]
Yep :)

**Pull request merged**